### PR TITLE
Add IAsyncEnumerable breaking change

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #26324 

_[Preview](https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/serialization/6.0/iasyncenumerable-serialization?branch=pr-en-us-26353)_

### CODE REVIEW

File: docs/core/compatibility/serialization/6.0/iasyncenumerable-serialization.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+Asynchronous serialization methods now enumerate any <xref:System.Collections.Generic.IAsyncEnumerable%601> instances in an object graph and then serialize them as JSON arrays. Synchronous serializations methods do not support <xref:System.Collections.Generic.IAsyncEnumerable%601> serialization and throw a <xref:System.NotSupportedException>.
+
+## Version introduced
+
+6.0 Preview 4
+
+## Type of breaking change
+
+This change can affect [binary compatibility](../../categories.md#binary-compatibility).
```
Comment (eiriktsarpalis) [inline docs/core/compatibility/serialization/6.0/iasyncenumerable-serialization.md:25]: The change doesn't impact binary compatibility, it is a behavior change affecting the IAE serialization format.

**Review by eiriktsarpalis — APPROVED** [review]
Other than the binary compatibility remark, LGTM.

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren once you respond to @eiriktsarpalis 's suggestion, you can :shipit:

File: docs/core/compatibility/serialization/6.0/iasyncenumerable-serialization.md (hunk 1 above)
Comment (gewarren) [reply docs/core/compatibility/serialization/6.0/iasyncenumerable-serialization.md:25]: @eiriktsarpalis Does it affect source compatibility then?

"Source code may encounter a breaking change in behavior when targeting the new runtime/component/SDK, such as compile errors or different run-time behavior."

Comment (eiriktsarpalis) [reply docs/core/compatibility/serialization/6.0/iasyncenumerable-serialization.md:25]: The change impacts runtime serialization behavior, but there should be no compile-time errors or binary compatibility issues.

**Pull request merged**